source/navitiacommon/navitiacommon/models.py

```diff
@@ -401,6 +401,29 @@ def get_from_id_or_name(cls, id=None, name=None):
         else:
             raise Exception({'error': 'instance is required'}, 400)
 
+    def delete_dataset(self):
+        result = db.session.query(DataSet, Job) \
+            .join(Job) \
+            .join(Instance) \
```

Comment:
do you really need to do this join? you don't use any fields from instances

Reply:
'self.id' in the filter. Need testing to check the results

Reply:
Results are the same. Removed